src/app/category/category.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+export interface Category {
+  id?: number;
+  name: string;
+  description?: string;
+}
+
+@Injectable({ providedIn: 'root' })
+export class CategoryService {
+  private baseUrl = '/categories';
```

Comment:
This cannot work, the backend is on a different server. Check other services and their use the environment.API variable that points to the backend (localhost:8080 during local development but tfgfinder-gei-api.fly.dev during deployment). As an example you can use https://github.com/UdL-EPS-SoftArch/tfgfinder-gei/blob/main/src/app/interest/interest.service.ts, though my recommendation is to use the HateoasClient that makes things even easier, like in https://github.com/UdL-EPS-SoftArch/tfgfinder-gei/blob/main/src/app/proposal/proposal.service.ts